backend/assignments/1/src/main/java/org/example/CryptoTradingApplication.java

```
@@ -0,0 +1,202 @@
+
+        // Load traders
+        this.traders = loadTradersFromCSV("C:\\Users\\Asus\\kdu\\kdu-coursework\\backend\\assignments\\1\\src\\main\\resources\\traders.csv");
+    }
+
+private List<Coins> loadCoinsFromCSV(String filePath) {
+    List<Coins> coins = new ArrayList<>();
+    try (CSVReader reader = new CSVReader(new FileReader(filePath))) {
+        String[] headers = reader.readNext(); // Read the header row
+
+        // Map column indices for required fields
+        int rankIndex = getIndex("Rank", headers);
+        int nameIndex = getIndex("Name", headers);
+        int symbolIndex = getIndex("Symbol", headers);
+        int priceIndex = getIndex("Price", headers);
+        int circulatingSupplyIndex = getIndex("Circulating Supply", headers);
+
+        // Iterate over each row in the CSV file
+        String[] nextLine;
+        while ((nextLine = reader.readNext()) != null) {
+            // Assuming "Rank", "Name", "Symbol", "Price," and "Circulating Supply" columns exist in the CSV
+            int rank = Integer.parseInt(nextLine[rankIndex]);
+            String name = nextLine[nameIndex];
+            String symbol = nextLine[symbolIndex];
+            double price = Double.parseDouble(nextLine[priceIndex]);
+            long circulatingSupply = Long.parseLong(nextLine[circulatingSupplyIndex]);
+
+            // Create a new Coin object
+            Coins coinnew = new Coins(rank,symbol, name, price, circulatingSupply);
+            coins.add(coinnew);
+        }
+    } catch (IOException e) {
+        e.printStackTrace();
+    } catch (CsvValidationException e) {
+        throw new RuntimeException(e);
+    }
+    return coins;
+}
+
+    private int getIndex(String columnName, String[] headers) {
+        for (int i = 0; i < headers.length; i++) {
+            if (columnName.equals(headers[i])) {
+                return i;
+            }
+        }
+        return -1; // Column not found
```

lets throw an exception here